lib/tern.js

```diff
@@ -761,8 +766,13 @@
 
       if (!completions.length && query.guess !== false && objType && objType.guessProperties)
         objType.guessProperties(function(p, o, d) {if (p != prop && p != "✖") gather(p, o, d);});
-      if (!completions.length && word.length >= 2 && query.guess !== false)
-        for (var prop in srv.cx.props) gather(prop, srv.cx.props[prop][0], 0);
+      if (!completions.length && word.length >= 2 && query.guess !== false) {
+        Object.keys(srv.cx.props).forEach(function(prop) {
+          srv.cx.props[prop].forEach(function(type) {
+    	      gather(prop, type, 0);	
+    	    });
+    	  });
+      }
```

Comment:
Is it really a good idea to start adding guessed properties multiple times? That has the potential to create a lot of noise, for very little gain (we don't know the accurate anyway).